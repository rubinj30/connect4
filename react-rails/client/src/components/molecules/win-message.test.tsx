import React from 'react';
import { shallow } from 'enzyme';
import { WinMessage } from './win-message.component';

describe('WinMessage', () => {
    it('should render properly on black win', () => {
        const fixture = <WinMessage currentTurn={'B'} resetBoard={jest.fn()} />;
        const result = shallow(fixture);
        expect(result).toBeDefined();
        expect(result).toMatchSnapshot();
    });
    it('should render properly on red win', () => {
        const fixture = <WinMessage currentTurn={'R'} resetBoard={jest.fn()} />;
        const result = shallow(fixture);
        expect(result).toBeDefined();
        expect(result).toMatchSnapshot();
    });
    it('resetBoard function should run when the reset button is clicked', () => {
        const mockFunc = jest.fn();
        const fixture = <WinMessage currentTurn={'R'} resetBoard={mockFunc} />;
        const result = shallow(fixture);
        const resetBoard = result.find('.pointer');
        expect(mockFunc).toHaveBeenCalledTimes(0);
        resetBoard.simulate('click');
        expect(mockFunc).toHaveBeenCalledTimes(1);
    });
});
