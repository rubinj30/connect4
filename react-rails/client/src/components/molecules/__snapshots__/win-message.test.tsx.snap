// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinMessage should render properly on black win 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WinMessage
    currentTurn="B"
    resetBoard={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="pb2 w-100 tc"
        >
          BLACK
           wins!
        </span>,
        <div
          className="pa2 br2 tc o-100 hover-bg-blue hover-white f5 bg-white pointer black"
          onClick={[MockFunction]}
        >
          Reset Board
        </div>,
      ],
      "className": "bg-black z-5 o-60 br3 absolute w-60 h-50 flex flex-column items-center justify-center pa5 f3 winMsg white ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "BLACK",
            " wins!",
          ],
          "className": "pb2 w-100 tc",
        },
        "ref": null,
        "rendered": Array [
          "BLACK",
          " wins!",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset Board",
          "className": "pa2 br2 tc o-100 hover-bg-blue hover-white f5 bg-white pointer black",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Reset Board",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="pb2 w-100 tc"
          >
            BLACK
             wins!
          </span>,
          <div
            className="pa2 br2 tc o-100 hover-bg-blue hover-white f5 bg-white pointer black"
            onClick={[MockFunction]}
          >
            Reset Board
          </div>,
        ],
        "className": "bg-black z-5 o-60 br3 absolute w-60 h-50 flex flex-column items-center justify-center pa5 f3 winMsg white ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "BLACK",
              " wins!",
            ],
            "className": "pb2 w-100 tc",
          },
          "ref": null,
          "rendered": Array [
            "BLACK",
            " wins!",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset Board",
            "className": "pa2 br2 tc o-100 hover-bg-blue hover-white f5 bg-white pointer black",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Reset Board",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`WinMessage should render properly on red win 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WinMessage
    currentTurn="R"
    resetBoard={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="pb2 w-100 tc"
        >
          RED
           wins!
        </span>,
        <div
          className="pa2 br2 tc o-100 hover-bg-blue hover-white f5 bg-white pointer red"
          onClick={[MockFunction]}
        >
          Reset Board
        </div>,
      ],
      "className": "bg-red z-5 o-60 br3 absolute w-60 h-50 flex flex-column items-center justify-center pa5 f3 winMsg white ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "RED",
            " wins!",
          ],
          "className": "pb2 w-100 tc",
        },
        "ref": null,
        "rendered": Array [
          "RED",
          " wins!",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset Board",
          "className": "pa2 br2 tc o-100 hover-bg-blue hover-white f5 bg-white pointer red",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Reset Board",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="pb2 w-100 tc"
          >
            RED
             wins!
          </span>,
          <div
            className="pa2 br2 tc o-100 hover-bg-blue hover-white f5 bg-white pointer red"
            onClick={[MockFunction]}
          >
            Reset Board
          </div>,
        ],
        "className": "bg-red z-5 o-60 br3 absolute w-60 h-50 flex flex-column items-center justify-center pa5 f3 winMsg white ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "RED",
              " wins!",
            ],
            "className": "pb2 w-100 tc",
          },
          "ref": null,
          "rendered": Array [
            "RED",
            " wins!",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset Board",
            "className": "pa2 br2 tc o-100 hover-bg-blue hover-white f5 bg-white pointer red",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Reset Board",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
